runtime/etc/debug/chpl_lldb_pretty_print.py

```diff
@@ -0,0 +1,543 @@
+            return list(self.synthetic_children.keys()).index(name)
+        else:
+            return -1
+
+    def get_child_at_index(self, index):
+        if index < 0 or index >= self.num_children():
+            return None
+        key = list(self.synthetic_children.keys())[index]
+        child_obj = self.synthetic_children[key]
+        return self.valobj.CreateValueFromAddress(
+            key, child_obj.GetLoadAddress(), child_obj.GetType()
+        )
+
+
+#
+# homogenous tuples
+#
+homogeneous_tuple_regex_c = re.compile(
+    r"^_tuple_([0-9]+)_star_([a-zA-Z0-9_]+)_chpl$"
+)
+homogeneous_tuple_regex_llvm = re.compile(r"^$")  # TODO
+
+
+def HomogeneousTupleRecognizer(sbtype, internal_dict):
+    typename = sbtype.GetName()
+    match = homogeneous_tuple_regex_c.match(
+        typename
+    ) or homogeneous_tuple_regex_llvm.match(typename)
+    return match is not None
+
+
+def HomogeneousTupleSummary(valobj, internal_dict):
+    typename = valobj.GetTypeName()
+    match = homogeneous_tuple_regex_c.match(
+        typename
+    ) or homogeneous_tuple_regex_llvm.match(typename)
+    if not match:
+        print(
+            f"HomogeneousTuple_Summary: type name '{typename}' did not match expected pattern"
+        )
+        return None
+
+    size = match.group(1)
+    vals = []
+    for i in range(int(size)):
+        child = valobj.GetNonSyntheticValue().GetChildAtIndex(i)
+        vals.append(child.GetSummary() or child.GetValue() or "")
+    return f"{', '.join(vals)}"
+
+
+class HomogeneousTupleProvider:
+    def __init__(self, valobj, internal_dict):
+        self.valobj = valobj
+
+        self.synthetic_children = {}
+        typename = self.valobj.GetTypeName()
+        match = homogeneous_tuple_regex_c.match(
+            typename
+        ) or homogeneous_tuple_regex_llvm.match(typename)
+        if not match:
+            print(
+                f"HomogeneousTupleProvider: type name '{typename}' did not match expected pattern"
+            )
+            return
+
+        size = int(match.group(1))
+        for i in range(size):
+            self.synthetic_children[str(i)] = self.valobj.GetChildAtIndex(i)
+
+    def has_children(self):
+        return self.num_children() > 0
+
+    def num_children(self):
+        return len(self.synthetic_children)
+
+    def get_child_index(self, name):
+        if name in self.synthetic_children:
+            return list(self.synthetic_children.keys()).index(name)
+        else:
+            return -1
+
+    def get_child_at_index(self, index):
+        if index < 0 or index >= self.num_children():
+            return None
+        key = list(self.synthetic_children.keys())[index]
+        return self.valobj.CreateValueFromExpression(
+            f"[{key}]", self.synthetic_children[key].GetValue()
+        )
+
+
+# array
+# _array_DefaultRectangularArr_1_int64_t_one_int64_t_int64_t_chpl
+# rank, idx_type, stride_kind, element_type, idxSignedType
```

Comment:
There should also probably be a note here that we'll want to support other types of arrays as well